src/lib/common/sol-power-supply-impl-linux.c

```diff
@@ -181,37 +181,33 @@ _get_enum_prop(const char *name, enum sol_power_supply_prop prop,
     return 0;
 }
 
-static bool
+static enum sol_util_iterate_dir_reason
 _list_supplies_cb(void *data, const char *dir_path, struct dirent *ent)
 {
     struct list_data *ldata = data;
     enum sol_power_supply_type type;
     char *name;
     int r;
 
-    if ((!strcmp(ent->d_name, ".")) || (!strcmp(ent->d_name, ".."))) {
-        return false;
-    }
-
     if (ldata->check_type) {
         r = sol_power_supply_get_type(ent->d_name, &type);
-        SOL_INT_CHECK(r, < 0, false);
+        SOL_INT_CHECK(r, < 0, SOL_UTIL_ITERATE_DIR_CONTINUE);
 
         if (type != ldata->type)
-            return false;
+            return SOL_UTIL_ITERATE_DIR_CONTINUE;
     }
 
     name = strdup(ent->d_name);
-    SOL_NULL_CHECK(name, false);
+    SOL_NULL_CHECK(name, -ENOMEM);
 
     r = sol_ptr_vector_append(ldata->list, name);
     if (r < 0) {
         SOL_WRN("Failed to append supply to list %s", name);
         free(name);
-        return false;
+        return r;
```

Comment:
I'm not sure that this and previous should stop looping... maybe previous code hopefully tried next directory entry... but if we are out of memory, maybe there's nothing much left to do...